db/entities/attempt.go

```diff
@@ -5,22 +5,44 @@ import (
 	"encoding/json"
 )
 
+type Attempt struct {
```

Comment:
`Attempt` did not have any fields related to the workspace, how can different workspaces be isolated?